extras/rxjava/src/main/java/org/asynchttpclient/extras/rxjava/AsyncHttpObservable.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2015 Target, Inc. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package org.asynchttpclient.extras.rxjava;
+
+import org.asynchttpclient.AsyncCompletionHandler;
+import org.asynchttpclient.BoundRequestBuilder;
+import org.asynchttpclient.HttpResponseStatus;
+import org.asynchttpclient.Response;
+import rx.Observable;
+import rx.Subscriber;
+import rx.functions.Func0;
+import rx.subjects.ReplaySubject;
+
+/**
+ * Provide RxJava support for executing requests. Request can be subscribed to and manipulated as needed.
+ * @See <a href="https://github.com/ReactiveX/RxJava" />
+ */
+public class AsyncHttpObservable {
+
+    /**
+     * Observe a request execution and emit the full response no matter what.
+     *
+     * @param supplier
+     * @return The cold observable (must be subscribed to in order to execute).
+     */
+    public static Observable<Response> toObservable(final Func0<BoundRequestBuilder> supplier) {
+        return toObservable(false, supplier);
+    }
+
+    /**
+     * Observe a request execution and emit an error for http status error codes >= 400.
+     *
+     * @param abortOnErrorStatus
+     * @param supplier
+     * @return The cold observable (must be subscribed to in order to execute).
+     */
+    public static Observable<Response> toObservable(final Boolean abortOnErrorStatus, final Func0<BoundRequestBuilder> supplier) {
+
+        //Get the builder from the function
+        final BoundRequestBuilder builder = supplier.call();
+
+        //create the observable from scratch
+        return Observable.create(new Observable.OnSubscribe<Response>() {
+
+            @Override
+            public void call(final Subscriber<? super Response> subscriber) {
+                try {
+                    AsyncCompletionHandler<Void> handler = new AsyncCompletionHandler<Void>() {
+                        @Override
+                        public State onStatusReceived(HttpResponseStatus status) throws Exception {
+                            State state = super.onStatusReceived(status);
+                            if (abortOnErrorStatus) {
+                                int code = status.getStatusCode();
+                                if (code >= 400) {
```

Comment:
Once you subscribe to a response, you will be forced to read the entire response body. Unless there's something I'm not considering?

It almost seems that configuring status codes on which to abort should be the domain of either the client or the builder configuration, not the observable.